packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -163,6 +176,9 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
       queries: {
         get: toCamelCase(['get', typeName]),
         list: toCamelCase(['list', plurality(typeName, true)]),
+        ...((ctx as TransformerContextProvider).isProjectUsingDataStore()
+          ? { sync: toCamelCase(['sync', plurality(typeName, true)]) }
+          : undefined),
```

Comment:
We add the function on all interfaces the context uses we won't need the type casting.